src/EditorFeatures/Core/LanguageServer/Handlers/CodeActions/CodeActionResolveHandler.cs

```diff
@@ -68,8 +70,9 @@ public async Task<LSP.CodeAction> HandleRequestAsync(LSP.CodeAction codeAction,
 
             var options = _globalOptions.GetCodeActionOptionsProvider();
 
+            var codeActionsCache = context.GetRequiredLspService<CodeActionsCache>();
```

Comment:
I assume the cache is thread-safe?

Reply:
yeah - the cache implementation remains unchanged (must be thread safe).